addons/payment_adyen/models/payment.py

```diff
@@ -151,8 +151,8 @@ def adyen_form_generate_values(self, values):
                 'sessionValidity': tmp_date,
                 'resURL': urls.url_join(base_url, AdyenController._return_url),
                 'merchantReturnData': json.dumps({'return_url': '%s' % values.pop('return_url')}) if values.get('return_url') else False,
-                'merchantSig': self._adyen_generate_merchant_sig('in', values),
             })
+            values['merchantSig'] = self._adyen_generate_merchant_sig('in', values)
```

Comment:
Hello, could you tell me why has this line been modified? I don't really think it needs to.

Otherwise your fix looks correct :+1:

Reply:
Hello @tbe-odoo ,
_adyen_generate_merchant_sig Method requires the latest computed data from adyen_form_generate_values method.
as you can see in sign generation method if you directly pass value while updating it, it will pass only business parameters and we don't need it. :)